File: javalang/src/SRP/problem/Employee.java
package SRP.problem;

public class Employee {
	private String employeeID;         
	private String employeeName;
	private String employeeAddress;
	private String contactNumber;
	private String employeeType;

	public void save() {
		//BD implementation here
		System.out.println("Saving in the Terminal .... " + this.getEmployeeID());
	}
	public void calculateTax() {

		if (this.getEmployeeType().equals("fulltime")) {
			// Task calc for full time employee
		}
		if (this.getEmployeeType().equals("contract")) {
			// Task calc for contractors
		}
	}
	public String getEmployeeID() {
		return this.employeeID;
	}	
	public void setEmployeeID(String ID) {
		this.employeeID=ID;
	}
	public String getEmployeeName() {
		return this.employeeName;
	}
	public void setEmployeeName(String employeeName) {
		this.employeeName = employeeName;
	}
	public String getEmployeeAddress() {
		return this.employeeAddress;
	}
	public void SetEmployeeAddress(String employeeAddress) {
		this.employeeAddress = employeeAddress;
	}
	public String GetContactNumber() {
		return this.contactNumber;
	}
	public void SetContactNumber(String contactNumber) {
		this.contactNumber = contactNumber;
	}
	public String getEmployeeType() {
		return this.employeeType;
	}
	public void SetEmployeeType(String employeeType) {
		this.employeeType = employeeType;
	}
}

File: golang/SRP/problem/employee/employee.go
package employee_problem

import "fmt"

type Employee struct {
	employeeID      string
	employeeName    string
	employeeAddress string
	contactNumber   string
	employeeType    string
}

func (e *Employee) Save() {
	//BD implementation here
	fmt.Println("Saving in the Terminal .... " + e.GetEmployeeID())
}

func (e *Employee) CalculateTax() {
	if e.GetEmployeeType() == "fulltime" {
		// Task calc for full time employee
	}
	if e.GetEmployeeType() == "contract" {
		// Task calc for contractors
	}
}
func (e *Employee) GetEmployeeID() string {
	return e.employeeID
}
func (e *Employee) SetEmployeeID(ID string) {
	e.employeeID = ID
}
func (e *Employee) GetEmployeeName() string {
	return e.employeeName
}
func (e *Employee) SetEmployeeName(employeeName string) {
	e.employeeName = employeeName
}
func (e *Employee) GetEmployeeAddress() string {
	return e.employeeAddress
}
func (e *Employee) SetEmployeeAddress(employeeAddress string) {
	e.employeeAddress = employeeAddress
}
func (e *Employee) GetContactNumber() string {
	return e.contactNumber
}
func (e *Employee) SetContactNumber(contactNumber string) {
	e.contactNumber = contactNumber
}
func (e *Employee) GetEmployeeType() string {
	return e.employeeType
}
func (e *Employee) SetEmployeeType(employeeType string) {
	e.employeeType = employeeType
}
